src/main/kotlin/Main.kt

```diff
@@ -77,6 +117,11 @@ fun runConsumer(q: Queue) =
         it
             .map { it.value() }
             .map { readScheduledTask(it) }
+            .advanceIf { !it.isExpired() }
```

Comment:
This is not a bug but might lead to "unexpected behaviour" since the expiry date refers to the last time a repeating polling might be starting rather than the last time a poll might occur which might result in weirdness on long intervals.

Reply:
Whoops, 

@Torbacka make sure that we check for expiry before publishing

Reply:
Think I manage to fix the problem.